src/http-loader.ts

```diff
@@ -11,6 +14,27 @@ export class TranslateHttpLoader implements TranslateLoader {
      * @returns {any}
      */
     public getTranslation(lang: string): any {
-        return this.http.get(`${this.prefix}${lang}${this.suffix}`);
+        if (this.loadedTranslations != null && this.loadedTranslations[lang] != null) {
+            return Observable.of(this.loadedTranslations[lang]);
+        }
+        return Observable.fromPromise(this.preLoad(lang));
+    }
+
+    /**
+     * Gets the translations from the server as Promise
+     * @param lang
+     * @returns Promise<any>
+     */
+    public preLoad(lang: string): Promise<any> {
+        return new Promise((resolve, reject) => {
+                this.http.get(`${this.prefix}${lang}${this.suffix}`)
```

Comment:
I think you can change the signature to:

```typescript
get().subscribe(
result => ()....,
error => ()....
);

```

Reply:
in this case better to use:
```js
    public preLoad(lang: string): Promise<any> {
        return this.httpClient.get(`${this.prefix}${lang}${this.suffix}`)
            .toPromise()
            .then(result => {
                this.loadedTranslations[lang] = result;
                return result;
            })
            .catch(() => null);
    }
```

Reply:
agree